sdk/trace/src/main/java/io/opentelemetry/sdk/trace/TracerSharedState.java

```diff
@@ -81,18 +81,20 @@ SpanProcessor getActiveSpanProcessor() {
   }
 
   /**
-   * Returns {@code true} if tracing is stopped.
+   * Returns {@code true} if tracing has been shut down.
    *
-   * @return {@code true} if tracing is stopped.
+   * @return {@code true} if tracing has been shut down.
    */
-  boolean isStopped() {
+  boolean hasBeenShutdown() {
+    // todo: do we really need a lock around this check? This is on the hot path of tracing.
```

Comment:
We needed it when we had two checks, now we only have one - so yeah let's delete this lock now

Reply:
Suggest separate PR since that is an internal implementation detail, this is a behavior change

Reply:
Point being this is urgent (before 1.0) the lock improvement can wait :)

Reply:
- Our 1.0 is about API stability, we generally wouldn't consider this PR as required-for-ga anyways, and definitely not required-for-next-rc

- It's a two line change that is a direct result of this PR, separating doesn't make sense it just decouples the cause/effect

Reply:
@breedx-splk and I have been working separately on getting rid of the lock. The question we have is... how important is it that calls to `shutdown()` are idempotent? That is, how much safety do we need to provide around not calling shutdown more than once on the processors/exporters?

Reply:
The lock on write, or a AtomicReference CAS seems important, but for this read we're just reading a volatile variable. Perhaps that's the confusion - it seems the variable was incorrectly marked volatile despite being guarded :) It's why I thought it's a no-brainer to remove the lock here.

Reply:
ok, done. It's still a volatile read in the hot path, but I guess that's probably better than a volatile read, guarded by a lock on the hot path. :)